_posts/SieveOfEratosthenes.md

```diff
@@ -0,0 +1,132 @@
+
+title: "Sieve of Eratosthenes - An ancient algorithm for prime numbers upto any limit"
+description: "This blog post is about Sieve of Eratosthenes which is an algorithm for finding all the prime numbers in a segment [1;n] using O(nloglogn) operations."
+category: dsa
+
+keywords: 
+    - Mentro
+   
+author: aastha 
+permalink: /dsa/:title/
+tags:
+    - prime number algorithm
+    - dsa
+    - sieve of eratosthenes
+---
+
+
+## Introduction
+t is easy to find if some number (say N) is prime or not — you simply need to check if at least one number from numbers lower or equal sqrt(n) is divisor of N. This can be achieved by simple code:
+
+boolean isPrime( int n ) {
+    if ( n == 1 ) return false; // by definition, 1 is not prime number
+    if ( n == 2 ) return true; // the only one even prime
+    for ( int i = 2; i * i <= n; ++i )
+        if ( n%i == 0 ) return false;
+    return true;
+}
+So it takes sqrt(n) steps to check this. Of course you do not need to check all even numbers, so it can be "optimized" a bit:
+
+boolean isPrime( int n ) {
+    if ( n == 1 ) return false; // by definition, 1 is not prime number
+    if ( n == 2 ) return true; // the only one even prime
+    if ( n%2 == 0 ) return false; // check if is even
+    for ( int i = 3; i * i <= n; i += 2 ) // for each odd number
+        if ( n%i == 0 ) return false;
+    return true;
+}
+So let say that it takes 0,5*sqrt(n) steps. That means it takes 500.000.000 steps to check that 1.000.000.007 is a prime.
+
+But there is great idea — why to "do not check even numbers". This idea can be extended to: "do not divide N by candidate numbers, but mark the prime multiples as 'not prime'".
+
+In other words: if we know about some number p, taht it is prime, we mark all multiples (2*p, 3*p, 4*p, ...) as not a prime. To implement this we need limit (max number we care about), let say it is 120 (because there is nice animation of algorithm on wikipedia).
```

Comment:
In other words: if we know about some number p, **that** it is prime, we mark all multiples (2*p, 3*p, 4*p, ...) as not a prime. To implement this we need a limit (max number we care about), let say it is 120 (because there is a nice animation of the algorithm on Wikipedia). - plz run your blog through Grammarly once it has a lot of grammatical mistakes

Reply:
(because there is a nice animation of the algorithm on Wikipedia) - this line is not needed